packages/git/src/browser/git-repository-provider.ts

```diff
@@ -71,9 +73,7 @@ export class GitRepositoryProvider {
             this.lazyRefresh();
         });
 
-        this._selectedRepository = await this.storageService.getData<Repository | undefined>(this.selectedRepoStorageKey);
```

Comment:
Why did you remove it? It is done that git does not empty UI before actual state is loaded.

Reply:
`ScmRepository` is a class, so it won't be serialized in the `StorageService`

Reply:
Could we jsonize scmrepositor data and recover classes on reload from such data?

Reply:
`ScmRepository` consumes handlers: https://github.com/theia-ide/theia/blob/71f277be8505b8e405e945854e07edc6c12f1200/packages/scm/src/browser/scm-service.ts#L201 that are not serialized

Reply:
ok, will have a look what can be done